xsd-core/src/main/java/io/ballerina/xsd/core/visitor/XSDVisitorImpl.java

```diff
@@ -0,0 +1,613 @@
+        return choiceName;
+    }
+
+    private static void generateSequenceType(StringBuilder stringBuilder,
+                                             StringBuilder childNodeBuilder, String sequenceName) {
+        stringBuilder.append(PUBLIC).append(WHITESPACE).append(TYPE).append(WHITESPACE).append(sequenceName);
+        stringBuilder.append(WHITESPACE).append(RECORD).append(WHITESPACE).append(OPEN_BRACES).append(VERTICAL_BAR);
+        stringBuilder.append(childNodeBuilder);
+        stringBuilder.append(VERTICAL_BAR).append(CLOSE_BRACES).append(SEMICOLON);
+    }
+
+    private void processChildNodes(boolean isOptional, NodeList childNodes,
+                                   StringBuilder stringBuilder) throws Exception {
+        int order = 1;
+        for (Node childNode : asIterable(childNodes)) {
+            Optional<XSDComponent> component = XSDFactory.generateComponents(childNode);
+            if (component.isEmpty()) {
+                continue;
+            }
+            component.get().setSubType(true);
+            component.get().setOptional(isOptional);
+            stringBuilder.append(addNamespace(this, getTargetNamespace()));
+            String orderAnnotation = XMLDATA_ORDER + WHITESPACE + OPEN_BRACES + VALUE + COLON + order + CLOSE_BRACES;
+            stringBuilder.append(orderAnnotation);
+            stringBuilder.append(component.get().accept(this));
+            order++;
+        }
+    }
+
+    private void processChildNode(boolean isOptional, Node childNode,
+                                  StringBuilder stringBuilder) throws Exception {
+        Optional<XSDComponent> component = XSDFactory.generateComponents(childNode);
+        if (component.isPresent()) {
+            component.get().setSubType(true);
+            component.get().setOptional(isOptional);
+            stringBuilder.append(addNamespace(this, getTargetNamespace()));
+            stringBuilder.append(component.get().accept(this));
+        }
+    }
+
+    private String applySequenceAnnotation(Node node, StringBuilder builder) {
+        Node maxOccurrenceNode = node.getAttributes().getNamedItem(MAX_OCCURS);
+        Node minOccurrenceNode = node.getAttributes().getNamedItem(MIN_OCCURS);
+        String maxOccurrence = (maxOccurrenceNode != null) ? maxOccurrenceNode.getNodeValue() : ONE;
+        String minOccurrence = (minOccurrenceNode != null) ? minOccurrenceNode.getNodeValue() : ONE;
+        String sequenceName = (nestedElements.containsKey(SEQUENCE_NAME))
+                ? Utils.resolveNameConflicts(SEQUENCE_NAME, nestedElements) : SEQUENCE_NAME;
+        sequenceName = (sequenceName.contains(COLON))
+                ? sequenceName.substring(sequenceName.indexOf(COLON) + 1) : sequenceName;
+        builder.append(XMLDATA_SEQUENCE).append(WHITESPACE).append(OPEN_BRACES);
+        builder.append(MIN_OCCURS).append(COLON).append(minOccurrence).append(COMMA);
```

Comment:
Dont we need a space in the midddle.
`minOccurs: 1` (space between `:` and `1`

Reply:
This is not needed as these records are been formatted at the end which ultimately adds necessary whitespaces into the records